verticapy/performance/vertica/collection/collection_tables.py

```diff
@@ -0,0 +1,584 @@
+"""
+Copyright  (c)  2018-2024 Open Text  or  one  of its
+affiliates.  Licensed  under  the   Apache  License,
+Version 2.0 (the  "License"); You  may  not use this
+file except in compliance with the License.
+
+You may obtain a copy of the License at:
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless  required  by applicable  law or  agreed to in
+writing, software  distributed  under the  License is
+distributed on an  "AS IS" BASIS,  WITHOUT WARRANTIES
+OR CONDITIONS OF ANY KIND, either express or implied.
+See the  License for the specific  language governing
+permissions and limitations under the License.
+"""
+from abc import abstractmethod
+from enum import Enum
+from typing import Mapping
+
+
+class CollectionTable:
+    """
+    Parent class for tables created by query profile export.
+
+    Child classes are expected to implement abstract methods
+        - get_create_table_sql()
+        - get_create_projection_sql()
+    """
+
+    def __init__(self, table_name: str, table_schema: str, key: str) -> None:
+        self.name = table_name
+        self.schema = table_schema
+        self.key = key
+        self.import_prefix = "qprof_"
+        self.import_suffix = f"_{key}"
+        self.staging_suffix = "_staging"
+
+    def get_import_name_fq(self) -> str:
+        return self._get_import_name(fully_qualified=True)
+
+    def get_import_name(self) -> str:
+        return self._get_import_name(fully_qualified=False)
+
+    def get_super_proj_name_fq(self) -> str:
+        return f"{self._get_import_name(fully_qualified=True)}_super"
+
+    def _get_import_name(self, fully_qualified) -> str:
+        return (
+            f"{self.schema}.{self.import_prefix}{self.name}{self.import_suffix}"
+            if fully_qualified
+            else f"{self.import_prefix}{self.name}{self.import_suffix}"
+        )
+
+    def get_staging_name(self) -> str:
+        return f"{self.schema}.{self.import_prefix}{self.name}{self.staging_suffix}{self.import_suffix}"
+
+    def get_drop_table_sql(self) -> str:
+        # Drop table can be in the base class because it is
+        # DROP TABLE IF EXISTS {self.get_import_name_fq}
+        raise NotImplementedError(f"Drop table not implemented yet")
```

Comment:
Do I understand it correctly that it will be implemented in the base class later?

Reply:
Yes, we'll add it another PR.